spec/lib/vanagon/component/source/git_spec.rb

```diff
@@ -1,49 +1,53 @@
 require 'vanagon/component/source/git'
 
 describe "Vanagon::Component::Source::Git" do
-  let(:klass) { Vanagon::Component::Source::Git }
-  let(:url) { 'git://github.com/puppetlabs/facter' }
-  let(:ref_tag) { 'refs/tags/2.2.0' }
-  let(:bad_sha) { 'FEEDBEEF' }
-  let(:workdir) { ENV["TMPDIR"] || '/tmp' }
+  # before(:all) blocks are run once before all of the examples in a group
+  before :all do
+    @klass = Vanagon::Component::Source::Git
+    @url = 'git://github.com/puppetlabs/facter.git'
+    @ref_tag = 'refs/tags/2.2.0'
+    @bad_sha = 'FEEDBEEF'
+    @workdir = nil
+  end
 
-  after(:each) { %x(rm -rf #{workdir}/facter) }
+  # before(:each) blocks are run before each example
+  before :each do
+    allow(Git)
+      .to receive(:ls_remote)
+            .and_return(true)
+  end
 
   describe "#initialize" do
     it "raises error on initialization with an invalid repo" do
-      # this test has a spelling error for the git repo
-      # * this is on purpose *
-      expect { klass.new("#{url}l.git", ref: ref_tag, workdir: workdir) }
-        .to raise_error Vanagon::InvalidRepo
+      # Ensure initializing a repo fails without calling over the network
+      allow(Git)
+        .to receive(:ls_remote)
+              .and_return(false)
+
+      expect { @klass.new(@url, ref: @ref_tag, workdir: @workdir) }
+        .to raise_error(Vanagon::InvalidRepo)
     end
   end
 
   describe "#dirname" do
     it "returns the name of the repo" do
-      git_source = klass.new(url, ref: ref_tag, workdir: workdir)
+      git_source = @klass.new(@url, ref: @ref_tag, workdir: @workdir)
       expect(git_source.dirname)
         .to eq('facter')
     end
 
     it "returns the name of the repo and strips .git" do
```

Comment:
Ah, that's my fault. The second test used to mangle the repo URL (which did not include `.git`) to include `.git`. That makes sense, since the implementation in `Vanagon::Component::Source::Git` is:

```ruby
# The dirname to reference when building from the repo
#
# @return [String] the directory where the repo was cloned
def dirname
  File.basename(url.path, ".git")
end
```